guides/source/active_record_callbacks.md

```diff
@@ -837,58 +837,59 @@ lead to invalid data.
 [`upsert_all`]:
     https://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-upsert_all
 
-Suppressing Callbacks
----------------------
+Suppressing Saving
+------------------
 
-In certain scenarios, you may need to temporarily prevent certain callbacks from
-being executed within your Rails application. This can be useful when you want
-to skip specific actions during certain operations without permanently disabling
-the callbacks.
+In certain scenarios, you may need to temporarily prevent records from being saved within your Rails application.
+This can be useful to skip the callbacks that would run if the record is saved successfully.
 
-Rails provides a mechanism for suppressing callbacks using the
-[`ActiveRecord::Suppressor`
-module](https://api.rubyonrails.org/classes/ActiveRecord/Suppressor.html). By
-using this module, you can wrap a block of code where you want to suppress
-callbacks, ensuring that they are not executed during that specific operation.
+Rails provides a mechanism to prevent saving records using the [`ActiveRecord::Suppressor` module]
+(https://api.rubyonrails.org/classes/ActiveRecord/Suppressor.html).
+By using this module, you can wrap a block of code where you want to avoid saving records and running their save callbacks.
 
-Let's consider a scenario where we have a `User` model with a callback that
-sends a welcome email to new users after they sign up. However, there might be
-cases where we want to create a user without sending the welcome email, such as
-during seeding the database with test data.
+Let's consider a scenario where we have a `User` `has_one` `Notification`.
+Creating a `User` will automatically create a `Notification` record as well.
+Saving a `Notification`, in turn, will trigger a callback that sends an email.
 
 ```ruby
 class User < ApplicationRecord
-  after_create :send_welcome_email
+  has_one :notification
+  after_create :create_notification
+end
+
+class Notification < ApplicationRecord
+  belongs_to :user
+
+  after_save :send_email
```

Comment:
I would just delete this callback as it makes the example unnecessarily complex.
All that matters is that we are suppressing the saving of the Notification.

Reply:
Hi p8 👋 Thank you for taking a look.

I think I structured the example this way because it's part of the "Active Record Callbacks" guide below the "Skipping Callbacks" section. If I remove the callback from `Notification`, the example would not skip an _explicit_ callback.

I think removing the callback in Notifications would make it simpler to explain the behavior of the `Suppressor`, I just thought the focus of this section of the guide is how to "skip callbacks" (with the suppressor).